seed-server/network-connection.sh

```diff
@@ -62,108 +67,142 @@ node_network="${node_network:-}"
-openvpn_network_netmask=$(CIDR2Netmask $openvpn_network)
+keepalive 10 60
 
-service_network_address=$(echo $service_network | cut -f1 -d/)
-service_network_netmask=$(CIDR2Netmask $service_network)
+# client-config-dir to push client specific configuration
+client-config-dir /client-config-dir
 
-pod_network_address=$(echo $pod_network | cut -f1 -d/)
-pod_network_netmask=$(CIDR2Netmask $pod_network)
+key "/srv/secrets/vpn-server/tls.key"
+cert "/srv/secrets/vpn-server/tls.crt"
+ca "/srv/secrets/vpn-server/ca.crt"
+dh "/srv/secrets/dh/dh2048.pem"
 
-sed -e "s/\${SERVICE_NETWORK_ADDRESS}/${service_network_address}/" \
-    -e "s/\${SERVICE_NETWORK_NETMASK}/${service_network_netmask}/" \
-    -e "s/\${POD_NETWORK_ADDRESS}/${pod_network_address}/" \
-    -e "s/\${POD_NETWORK_NETMASK}/${pod_network_netmask}/" \
-    -e "s/\${OPENVPN_NETWORK_ADDRESS}/${openvpn_network_address}/" \
-    -e "s/\${OPENVPN_NETWORK_NETMASK}/${openvpn_network_netmask}/" \
-    -e "s/\${POOL_START_IP}/${pool_start_ip}/" \
-    -e "s/\${POOL_END_IP}/${pool_end_ip}/" \
-    openvpn.config.template > openvpn.config
+tls-auth "/srv/secrets/tlsauth/vpn.tlsauth" 0
+EOF
 
-sed -e "s/\${SERVICE_NETWORK_ADDRESS}/${service_network_address}/" \
-    -e "s/\${SERVICE_NETWORK_NETMASK}/${service_network_netmask}/" \
-    -e "s/\${POD_NETWORK_ADDRESS}/${pod_network_address}/" \
-    -e "s/\${POD_NETWORK_NETMASK}/${pod_network_netmask}/" \
-    /client-config-dir/vpn-shoot-client.template > /client-config-dir/vpn-shoot-client
+# Write config that is dependent on the IP family
+if [[ "${IP_FAMILIES:-}" = "IPv4" ]]; then
+  {
+    printf 'proto tcp4-server\n'
+    printf 'server %s %s nopool\n' "$(echo $openvpn_network | cut -f1 -d/)" "$(CIDR2Netmask $openvpn_network)"
+    printf 'ifconfig-pool %s %s\n' "$pool_start_ip" "$pool_end_ip"
+  } >>openvpn.config
 
-if [[ -n $is_ha ]]; then
-    for ((i=0; i < $HA_VPN_CLIENTS; i++))
-    do
-        sed -e "s/\${SERVICE_NETWORK_ADDRESS}/${service_network_address}/" \
-            -e "s/\${SERVICE_NETWORK_NETMASK}/${service_network_netmask}/" \
-            -e "s/\${POD_NETWORK_ADDRESS}/${pod_network_address}/" \
-            -e "s/\${POD_NETWORK_NETMASK}/${pod_network_netmask}/" \
-            -e "s/\${LOCAL_ADDRESS}/192.168.123.$(( vpn_index * 64 + i + 2 ))/" \
-            -e "s/\${OPENVPN_NETWORK_NETMASK}/${openvpn_network_netmask}/" \
-            /client-config-dir/vpn-shoot-client-n.template > /client-config-dir/vpn-shoot-client-$i
-    done
+  {
+    printf 'iroute %s %s\n' "$(echo $service_network | cut -f1 -d/)" "$(CIDR2Netmask $service_network)"
+    printf 'iroute %s %s\n' "$(echo $pod_network | cut -f1 -d/)" "$(CIDR2Netmask $pod_network)"
+  } >/client-config-dir/vpn-shoot-client
+else
+  {
+    printf 'proto tcp6-server\n'
+    printf 'server-ipv6 %s\n' "$openvpn_network"
+  } >>openvpn.config
+
+  {
+    printf 'iroute-ipv6 %s\n' "$service_network"
+    printf 'iroute-ipv6 %s\n' "$pod_network"
+  } >/client-config-dir/vpn-shoot-client
```

Comment:
What was the reason to remove the templates and `printf` the whole config?
It introduces a big change that's hard to review.

Reply:
I believe the the template was really hard to work with considering our use-case so that's why we changed it.

But tim might have additional info...
CC: @timebertt

Reply:
Yeah, the template got ugly once we introduced the new cases for IPv6. Hence, we added a commit 865f5d39459d1b04c352945f4b9d9fed074435b9 which shouldn't change the output but only how the final config is calculated (template vs printf). 
You might find it helpful to review the PR commit by commit :)